### ISSUE TITLE
Python 3.8 fails to load SDL2.dll

### ISSUE THREAD

**Issue by einarf** [marcusva/py-sdl2#145]
For some reason pysdl2 fails to load `SDL2.dll` in python 3.8.

```
    try:
        dll = DLL("SDL2", ["SDL2", "SDL2-2.0"], os.getenv("PYSDL2_DLL_PATH"))
    except RuntimeError as exc:
>       raise ImportError(exc)
E       ImportError: found ['SDL2.dll'], but it's not usable for the library SDL2
```

I have tried with both x84 and x64 dll from libsdl.org:
```
SDL2-2.0.10-win32-x86
SDL2-2.0.10-win32-x64
```

This works fine in python 3.5, 3.6 and 3.7, so there must have been a change in python 3.8.

**Comment by a-hurst** [comment]
Definitely something that needs to be fixed, thanks for reporting the issue! I'll look into this.

**Comment by a-hurst** [comment]
This is now hopefully fixed in the latest commit (tested and working in a Windows 10 VM with the latest SDL2 and SDL2 mixer 64-bit DLLs). Try it again on your end and let me know if it works!

**Comment by marcusva** [comment]
@einarf: I cannot reproduce the behaviour, you are experiencing (without e004f6d), if PYSDL2_DLL_PATH is properly set or if the DLLs reside in a system directory. Can you please provide an example?

**Comment by einarf** [comment]
In my case I had SDL.dll in the root of my project (cwd). The commit e004f6d does make that possible again as it adds `./`

You are right that when having the dll in system32 it does work out of the box in py38 without the change. This was mainly about having the dll in cwd. Your change fixes that.

**Comment by marcusva** [comment]
Did you set PYSDL2_DLL_PATH to the root of your project?

**Comment by einarf** [comment]
Nope. Maybe I should just accept that this is the right things to do it 😄

**Comment by marcusva** [comment]
That would be highly appreciated. Please try it without that patch.

**Comment by einarf** [comment]
Works fine when setting `PYSDL2_DLL_PATH`` to `./` or `Path.cwd()`. for loading the local dll. I'm guessing this is to be expected in py38 as `PATH` is no longer used to look up ddls. Might just want to keep things unchanged as this is the expected behavior in py38.

What do you think?

**Comment by marcusva** [comment]
Using PYSDL2_DLL_PATH was always the preferred way to do things, especially since DLL detection, resolution paths and such are different from platform to platform and configured ACLs.

That being said, relying on PATH worked/works by chance, rather than being a supported way of doing things. The transition however will be annoying for users on Windows. 

@a-hurst: we should revert e004f6d and maybe add some additional warning message for Python 3.8 for the time being. What do you think?

**Comment by einarf** [comment]
I think a warning is a great idea as the "dll in cwd"-trick is something that has been ingrained in a lot people over the last two decades as valid solution.

**Comment by einarf** [comment]
Suggested added in #146

**Comment by a-hurst** [comment]
@marcusva this issue is the last thing other than the README I have on my "for sure address before 0.9.7" list. I haven't closed it yet because you weren't 100% on board with [the fix I committed to master](https://github.com/marcusva/py-sdl2/commit/e004f6ddd49b3b3c4a6836357ab98e6fdbf48f90) (should have been a PR, sorry), and were wondering whether pysdl2 should require that `PYSDL2_DLL_PATH` be explicitly set when not using system-installed libraries, with a helpful warning message informing people of the change. Is this still your preferred approach? If so, I'll remove my fix and add in the warning.

**Comment by a-hurst** [comment]
sdl2dll branch merged, pysdl2-dll wheels up on PyPI, pysdl2-dll information added to the README and docs. I think this can finally be closed!

**Comment by Baekalfen** [comment]
Any news on this? I'm seeing this both locally and in Travis https://travis-ci.com/github/Baekalfen/PyBoy/jobs/323501397

**Comment by a-hurst** [comment]
Hey @Baekalfen, unless Python's DLL loading policy on Windows has changed again since 3.8.0 this specific issue should be resolved: the only reason this issue is still open is because the original author of the project wasn't sure if the way we fixed it was an ideal solution. It looks like the issue you're encountering might be due to a different problem with your DLLs? To that end, are you using pysdl2 to install the Windows DLLs locally? If so I can rule out bad DLLs as the cause.

**Comment by Baekalfen** [comment]
Now that I think of it, it passes Cython compilation, and the error appears when I start the testing. Maybe it's not PySDL2. But I might be in the same boat as you were. How did you end up fixing it?

**Comment by a-hurst** [comment]
Basically, Python 3.8 changed how DLL loading with ctypes worked on Windows, such that you needed to specify the relative path to a DLL in the current working directory in order to get it to work (i.e if loading a dll, you needed to pass the path "./SDL2.dll" instead of just "SDL2.dll"). I just fixed it by adding a "./" prefix to the DLL name if it's in the current working directory, but I think you might be looking at a different issue given that you're using Cython.

Actually, this sounds like it might be a dynamic linking issue: when you're building PyBoy on Windows, are you linking SDL2 dynamically or statically? If the former, you might want to give static linking a try: I've looked into it a ton, and it seems like dynamic linking + compiled Python wheels is a huge mess that only people like the Numpy and Pillow maintainers have properly figured out how to do reliably, [since there's no easy standard way of doing it on Windows](https://discuss.python.org/t/need-auditwheel-like-utility-for-other-platforms/2028) like you have on Linux with auditwheel or macOS with delocate.
